extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -0,0 +1,216 @@
+    ref,
+    () => ({
+      open,
+      close,
+      toggle,
+    }),
+    [open, close, toggle]
+  );
+
+  useEffect(() => {
+    const sidebarElement = sidebarRef.current;
+    const maskElement = maskRef.current;
+    if (sidebarElement == undefined || maskElement == undefined) {
+      return;
+    }
+
+    let sidebarAnimation;
+    let maskAnimation;
+    // "Make Visible" Animation
+    if (opened) {
+      const postVisibleAnim = () => {
+        setStyle(sidebarElement, 'transform', 'translateX(0)');
+        setStyle(maskElement, 'opacity', 1);
+        sidebarElement./*REVIEW*/ focus();
+      };
+      if (!sidebarElement.animate || !maskElement.animate) {
+        postVisibleAnim();
+        return;
+      }
+      sidebarAnimation = sidebarElement.animate(
+        ANIMATION_PRESETS[side === 'left' ? 'slideInLeft' : 'slideInRight'],
+        {
+          duration: ANIMATION_DURATION,
+          fill: 'both',
+          easing: ANIMATION_EASE_IN,
+        }
+      );
+      sidebarAnimation.onfinish = postVisibleAnim;
+      maskAnimation = maskElement.animate(ANIMATION_PRESETS.fadeIn, {
+        duration: ANIMATION_DURATION,
+        fill: 'both',
+        easing: ANIMATION_EASE_IN,
+      });
+    } else {
+      // "Make Invisible" Animation
+      const postInvisibleAnim = () => {
+        if (onAfterCloseRef.current) {
+          onAfterCloseRef.current();
+        }
+        sidebarAnimation = null;
+        maskAnimation = null;
+        setMounted(false);
+      };
+      if (!sidebarElement.animate || !maskElement.animate) {
+        postInvisibleAnim();
+        return;
+      }
+      sidebarAnimation = sidebarElement.animate(
+        ANIMATION_PRESETS[side === 'left' ? 'slideInLeft' : 'slideInRight'],
+        {
+          duration: ANIMATION_DURATION,
+          direction: 'reverse',
+          fill: 'both',
+          easing: ANIMATION_EASE_IN,
+        }
+      );
+      sidebarAnimation.onfinish = postInvisibleAnim;
+      maskAnimation = maskElement.animate(ANIMATION_PRESETS.fadeIn, {
+        duration: ANIMATION_DURATION,
+        direction: 'reverse',
+        fill: 'both',
+        easing: ANIMATION_EASE_IN,
+      });
+    }
+    return () => {
+      if (sidebarAnimation) {
+        sidebarAnimation.cancel();
+      }
+      if (maskAnimation) {
+        maskAnimation.cancel();
+      }
+    };
+  }, [opened, onAfterCloseRef, side]);
+
+  return (
+    mounted && (
+      <>
+        <ContainWrapper
+          as={Comp}
+          ref={(r) => {
+            sidebarRef.current = r;
+          }}
+          size={false}
+          layout={true}
+          paint={true}
+          className={`${classes.baseClass} ${
+            side === 'left' ? classes.left : classes.right
+          }`}
+          role="menu"
+          tabindex="-1"
+          {...rest}
+        >
+          {children}
+        </ContainWrapper>
+        <div
+          ref={(r) => {
+            maskRef.current = r;
```

Comment:
Just `ref={maskRef}`?

Reply:
Thanks updated, I was meaning to ask, is there a reason the `ContainWrapper` above uses this type of syntax?

Reply:
I don't see any reason, tbh.

Reply:
Updated to `ref={sidebarRef}` in the lines above as well